boards.txt

```diff
@@ -2800,6 +2800,479 @@ aventen_s3_sync.menu.EraseFlash.all.upload.erase_cmd=-e
+BharatPi-LoRa.menu.PartitionScheme.huge_app=Huge APP (3MB No OTA/1MB SPIFFS)
+BharatPi-LoRa.menu.PartitionScheme.huge_app.build.partitions=huge_app
+BharatPi-LoRa.menu.PartitionScheme.huge_app.upload.maximum_size=3145728
+BharatPi-LoRa.menu.PartitionScheme.min_spiffs=Minimal SPIFFS (1.9MB APP with OTA/190KB SPIFFS)
+BharatPi-LoRa.menu.PartitionScheme.min_spiffs.build.partitions=min_spiffs
+BharatPi-LoRa.menu.PartitionScheme.min_spiffs.upload.maximum_size=1966080
+BharatPi-LoRa.menu.PartitionScheme.fatflash=16M Flash (2MB APP/12.5MB FATFS)
+BharatPi-LoRa.menu.PartitionScheme.fatflash.build.partitions=ffat
+BharatPi-LoRa.menu.PartitionScheme.fatflash.upload.maximum_size=2097152
+BharatPi-LoRa.menu.PartitionScheme.app3M_fat9M_16MB=16M Flash (3MB APP/9.9MB FATFS)
+BharatPi-LoRa.menu.PartitionScheme.app3M_fat9M_16MB.build.partitions=app3M_fat9M_16MB
+BharatPi-LoRa.menu.PartitionScheme.app3M_fat9M_16MB.upload.maximum_size=3145728
+BharatPi-LoRa.menu.PartitionScheme.rainmaker=RainMaker
+BharatPi-LoRa.menu.PartitionScheme.rainmaker.build.partitions=rainmaker
+BharatPi-LoRa.menu.PartitionScheme.rainmaker.upload.maximum_size=3145728
+BharatPi-LoRa.menu.PartitionScheme.custom=Custom
+BharatPi-LoRa.menu.PartitionScheme.custom.build.partitions=
+BharatPi-LoRa.menu.PartitionScheme.custom.upload.maximum_size=16777216
+
+BharatPi-LoRa.menu.CPUFreq.240=240MHz (WiFi/BT)
+BharatPi-LoRa.menu.CPUFreq.240.build.f_cpu=240000000L
+BharatPi-LoRa.menu.CPUFreq.160=160MHz (WiFi/BT)
+BharatPi-LoRa.menu.CPUFreq.160.build.f_cpu=160000000L
+BharatPi-LoRa.menu.CPUFreq.80=80MHz (WiFi/BT)
+BharatPi-LoRa.menu.CPUFreq.80.build.f_cpu=80000000L
+BharatPi-LoRa.menu.CPUFreq.40=40MHz (40MHz XTAL)
+BharatPi-LoRa.menu.CPUFreq.40.build.f_cpu=40000000L
+BharatPi-LoRa.menu.CPUFreq.26=26MHz (26MHz XTAL)
+BharatPi-LoRa.menu.CPUFreq.26.build.f_cpu=26000000L
+BharatPi-LoRa.menu.CPUFreq.20=20MHz (40MHz XTAL)
+BharatPi-LoRa.menu.CPUFreq.20.build.f_cpu=20000000L
+BharatPi-LoRa.menu.CPUFreq.13=13MHz (26MHz XTAL)
+BharatPi-LoRa.menu.CPUFreq.13.build.f_cpu=13000000L
+BharatPi-LoRa.menu.CPUFreq.10=10MHz (40MHz XTAL)
+BharatPi-LoRa.menu.CPUFreq.10.build.f_cpu=10000000L
+
+BharatPi-LoRa.menu.FlashMode.qio=QIO
+BharatPi-LoRa.menu.FlashMode.qio.build.flash_mode=dio
+BharatPi-LoRa.menu.FlashMode.qio.build.boot=qio
+BharatPi-LoRa.menu.FlashMode.dio=DIO
+BharatPi-LoRa.menu.FlashMode.dio.build.flash_mode=dio
+BharatPi-LoRa.menu.FlashMode.dio.build.boot=dio
+
+BharatPi-LoRa.menu.FlashFreq.80=80MHz
+BharatPi-LoRa.menu.FlashFreq.80.build.flash_freq=80m
+BharatPi-LoRa.menu.FlashFreq.40=40MHz
+BharatPi-LoRa.menu.FlashFreq.40.build.flash_freq=40m
+
+BharatPi-LoRa.menu.FlashSize.4M=4MB (32Mb)
+BharatPi-LoRa.menu.FlashSize.4M.build.flash_size=4MB
+BharatPi-LoRa.menu.FlashSize.8M=8MB (64Mb)
+BharatPi-LoRa.menu.FlashSize.8M.build.flash_size=8MB
+BharatPi-LoRa.menu.FlashSize.8M.build.partitions=default_8MB
+BharatPi-LoRa.menu.FlashSize.16M=16MB (128Mb)
+BharatPi-LoRa.menu.FlashSize.16M.build.flash_size=16MB
```

Comment:
If it also uses 4MB flash please remove those options and remove the unusable partitions.